### ISSUE TITLE
Invalid Nodelabel check on aks 1.16+

### ISSUE THREAD

**Issue by ncrothe** [Azure/AKS#1937]
**What happened**:

We are updating our aks cluster to 1.16 (from 1.15) and had to change our previous use of node labels kubernetes.io/lifecycle to node.kubernetes.io/lifecycle (in accordance with https://github.com/kubernetes/enhancements/blob/master/keps/sig-auth/0000-20170814-bounding-self-labeling-kubelets.md#proposal). However that is considered invalid by aks leading to the very confusing error:
```
Error: creating Managed Kubernetes Cluster "aks-k8s-staging" (Resource Group "rg-k8s-staging"): containerservice.ManagedClustersClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Co
de="BadRequest" Message="For Kubernetes version greater than or equal to 1.16, node label keys in the 'kubernetes.io' or 'k8s.io' namespace must begin with an allowed prefix (kubelet.kubernetes.io, node.kubernete
s.io) or be in the specifically allowed set (beta.kubernetes.io/arch, failure-domain.beta.kubernetes.io/zone, failure-domain.beta.kubernetes.io/region, failure-domain.kubernetes.io/region, kubernetes.io/os, kuber
netes.io/arch, beta.kubernetes.io/os, failure-domain.kubernetes.io/zone, kubernetes.io/hostname, kubernetes.io/instance-type, beta.kubernetes.io/instance-type). For more information, please check https://aka.ms/a
ks-node-labels-116." Target="node.kubernetes.io/lifecycle=on-demand"
```
(that is through Terraform and the azurerm provider, but similar errors happen with az cli).
Note the attempted label and the error message saying to use a label exacrtly like the one attempted.

**What you expected to happen**:

AKS allows node labels of the form [*.]node.kubernetes.io/*

**How to reproduce it (as minimally and precisely as possible)**:
Same problem with creating nodepools, so easiest to confirm with:
Run `az aks nodepool add --cluster-name <mycluster> --name <myname> --resource-group <myrg> --labels node.kubernetes.io/test=value`

**Environment**:
Kubernetes 1.16+, happens already on cluster creation.

**Comment by yangl900** [comment]
thanks for reporting this issue @ncrothe . I think that is a bug, we did it wrong. We allowed `foo.node.kubernetes.io` but not `node.kubernetes.io`. We'll get it fixed soon.

**Comment by yangl900** [comment]
The fix has been rolled out in AKS 20201109 release.
